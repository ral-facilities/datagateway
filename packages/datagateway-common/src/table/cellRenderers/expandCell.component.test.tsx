import React from 'react';
import { shallow } from 'enzyme';
import ExpandCell from './expandCell.component';

describe('Expand cell component', () => {
  const setExpandedIndex = jest.fn();
  const expandCellProps = {
    columnIndex: 1,
    dataKey: 'test',
    isScrolling: false,
    rowIndex: 1,
    rowData: '',
    expandedIndex: 1,
    setExpandedIndex,
  };

  afterEach(() => {
    setExpandedIndex.mockClear();
  });

  it('renders correctly when expanded', () => {
    const wrapper = shallow(<ExpandCell {...expandCellProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sets the expanded index to -1 when ExpandLess button is pressed', () => {
    const wrapper = shallow(<ExpandCell {...expandCellProps} />);

    wrapper.childAt(0).prop('onClick')();
    expect(setExpandedIndex).toHaveBeenCalledWith(-1);
  });

  it('renders correctly when not expanded', () => {
    const wrapper = shallow(
      <ExpandCell {...expandCellProps} expandedIndex={2} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('sets the expanded index to rowIndex when ExpandMore button is pressed', () => {
    const wrapper = shallow(
      <ExpandCell {...expandCellProps} expandedIndex={2} />
    );

    wrapper.childAt(0).prop('onClick')();
    expect(setExpandedIndex).toHaveBeenCalledWith(expandCellProps.rowIndex);
  });
});
