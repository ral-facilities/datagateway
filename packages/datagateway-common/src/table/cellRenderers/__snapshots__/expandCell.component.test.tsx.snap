// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expand cell component renders correctly when expanded 1`] = `
<ForwardRef(TableCell)
  component="div"
  padding="checkbox"
  size="small"
  variant="body"
>
  <ForwardRef(IconButton)
    aria-label="Hide details"
    onClick={[Function]}
    size="large"
  >
    <Memo(ForwardRef(ExpandLessIcon)) />
  </ForwardRef(IconButton)>
</ForwardRef(TableCell)>
`;

exports[`Expand cell component renders correctly when not expanded 1`] = `
<ForwardRef(TableCell)
  component="div"
  padding="checkbox"
  size="small"
  variant="body"
>
  <ForwardRef(IconButton)
    aria-label="Show details"
    className="tour-dataview-expand"
    onClick={[Function]}
    size="large"
  >
    <Memo(ForwardRef(ExpandMoreIcon)) />
  </ForwardRef(IconButton)>
</ForwardRef(TableCell)>
`;
